// bangla-calendar/lib/esm/utils.js
var convertNumbers = function (data) {
    var numbers = {
        0: '০',
        1: '১',
        2: '২',
        3: '৩',
        4: '৪',
        5: '৫',
        6: '৬',
        7: '৭',
        8: '৮',
        9: '৯',
        '.': '.',
        '-': '-'
    };
    var result = '';
    if (!data) {
        return '';
    }
    var input = data.toString();
    var length = input.length;
    // eslint-disable-next-line no-plusplus
    for (var i = 0; i < length; i++) {
        if (Number.isNaN(parseFloat(input[i])) || Number.isNaN(input[i] - 0)) {
            result += input[i];
        }
        else {
            result += numbers[input[i]];
        }
    }
    return result;
};
var banglaMonths = [
    'বৈশাখ',
    'জ্যৈষ্ঠ',
    'আষাঢ়',
    'শ্রাবণ',
    'ভাদ্র',
    'আশ্বিন',
    'কার্তিক',
    'অগ্রহায়ণ',
    'পৌষ',
    'মাঘ',
    'ফাল্গুন',
    'চৈত্র',
];
var banglaWeekDays = [
    'রবি',
    'সোম',
    'মঙ্গল',
    'বুধ',
    'বৃহস্পতি',
    'শুক্র',
    'শনি',
];
var isLeapYear = function (year) {
    if (year === void 0) { year = 0; }
    return year % 100 === 0 ? year % 400 === 0 : year % 4 === 0;
};
var isValidDate = function (date) {
    if (Object.prototype.toString.call(date) === '[object Date]') {
        // eslint-disable-next-line no-restricted-globals
        if (isNaN(date)) {
            return false;
        }
        return true;
    }
    return false;
};
var formatDay = function (day, format) {
    if (day === void 0) { day = 1; }
    if (format === void 0) { format = 'D'; }
    var d = day.toString();
    switch (format) {
        case 'DD':
            d = d.length === 1 ? "0".concat(d) : d;
            return convertNumbers(d);
        default:
            return convertNumbers(day);
    }
};
var formatWeekDay = function (day, format) {
    if (day === void 0) { day = 0; }
    if (format === void 0) { format = 'eeee'; }
    switch (format) {
        case 'eee':
            return banglaWeekDays[day];
        default:
            return "".concat(banglaWeekDays[day], "\u09AC\u09BE\u09B0");
    }
};
var formatMonth = function (month, format) {
    if (month === void 0) { month = 0; }
    if (format === void 0) { format = 'MMMM'; }
    var m = (month + 1).toString();
    switch (format) {
        case 'M':
            return convertNumbers(m);
        case 'MM':
            m = m.length === 1 ? "0".concat(m) : m;
            return convertNumbers(m);
        default:
            return banglaMonths[month];
    }
};
var formatYear = function (year, format) {
    if (year === void 0) { year = 0; }
    if (format === void 0) { format = 'YYYY'; }
    var y = year.toString();
    switch (format) {
        case 'YY':
            return convertNumbers(y.substring(y.length - 2));
        case 'YYYYb':
            return "".concat(convertNumbers(y), " (\u09AC\u0999\u09CD\u0997\u09BE\u09AC\u09CD\u09A6)");
        default:
            return convertNumbers(y);
    }
};
var getJulianDate = function (year, month, day) {
    var y = year;
    var m = month;
    if (month <= 2) {
        y -= 1;
        m += 12;
    }
    var A = Math.floor(y / 100);
    var B = 2 - A + Math.floor(A / 4);
    var JD = Math.floor(365.25 * (y + 4716)) +
        Math.floor(30.6001 * (m + 1)) +
        day +
        B -
        1524.5;
    return JD;
};
var monthLengthIN = [
    0, 30.93081, 62.35364, 93.9999999999999, 125.47636, 156.48933, 186.92405,
    216.3179999, 246.3153999, 275.14288, 305.09428, 334.91145, 365.258756,
];
var startJulianDate = 1938094.4629;
var yearLength = 365.258756;
var errorMessage = 'Invalid Date';

export { banglaMonths, banglaWeekDays, convertNumbers, errorMessage, formatDay, formatMonth, formatWeekDay, formatYear, getJulianDate, isLeapYear, isValidDate, monthLengthIN, startJulianDate, yearLength };
